opentelemetry-sdk/src/logs/log_emitter.rs

```diff
@@ -97,25 +100,46 @@ impl LoggerProvider {
     }
 
     /// Force flush all remaining logs in log processors and return results.
-    pub fn force_flush(&self) -> Vec<LogResult<()>> {
-        self.log_processors()
-            .iter()
-            .map(|processor| processor.force_flush())
-            .collect()
+    pub fn force_flush(&self) -> LogResult<()> {
```

Comment:
TracerProvider returns a list of results from processors. If there a reason why we prefer single result over multiple results here? 

IMO it only helps simplify the use case where users want to fail when any processor fails. But it's relatively easy for users to do it on their own (Just `.iter().filter().is_empty()` should be enough)

However, it will make it harder for users that care about one processor over another to decide whether the force_flush is worth retrying. For example, let's say I have a simple processor that print to stdout and a batch processor to print to backend, with simple processor mostly used for debugging purpose. Then with this change I need to prase the error results to determine if the error comes from batch processor. 

I also don't think performance will improves just by chaning the return types but let me know if you think it will

Reply:
This is not done for any performance reasons, but simply to ensure consistency.
`MeterProvider` always returns single Result, and not a vec of Results from each of its readers. This PR makes `LoggerProvider` consistent with metrics.
(Neither is consistent with Traces, as Traces do have global shutdown, but others dont have that. However traces don't have shutdown on the provider itself.)

> However, it will make it harder for users that care about one processor over another to decide whether the force_flush is worth retrying. For example, let's say I have a simple processor that print to stdout and a batch processor to print to backend, with simple processor mostly used for debugging purpose. Then with this change I need to prase the error results to determine if the error comes from batch processor.

Agree this not very ideal.

I can revert changes to `force_flush` from this PR, as it probably need more discussion and alignment.

Reply:
Sounds good. Let's open an issue then?